### ISSUE TITLE
Keyboard didn't work in some games (But seems already fixed, only to be known)

### ISSUE THREAD

**Issue by LukasThyWalls** [adventuregamestudio/ags#2597]
Hello.

I want to report this issue, although It seems Its fixed some how already, because It was very weird and a bit tricky to figure out what was happening, and I didn't find any mention anywhere for this or something similiar (I have searched here in [Issues](https://github.com/adventuregamestudio/ags/issues), [Commits](https://github.com/adventuregamestudio/ags/commits/master/) and in the [Forums](https://www.adventuregamestudio.co.uk/forums/)). I don't know If there is somewhere to put "known issues" like this one, because It affects several known games made by different people, and seems and unknown issue of AGS, although somehow is already fixed. 

**Describe the issue**

Simply with some games the keyboard did not work. Any keystroke. Mouse works fine.

**Games** and **AGS Versions**

- _Billy Masters was Right_ ([Itch](https://postmodernadventures.itch.io/billy)) [_AGS 3.4.3.1_]
- _Dead Dimension_ ([Itch](https://postmodernadventures.itch.io/dead-dimension)) [_AGS 3.4.1.15_]
- _If On A Winter's Night, Four Travelers_ ([Itch](https://laurahunt.itch.io/if-on-a-winters-night-four-travelers) & [Steam](https://store.steampowered.com/app/1603980/If_On_A_Winters_Night_Four_Travelers/)) [_AGS 3.5.0.31_]

**Steps to Reproduce**

There is no steps because the games simply the keyboard didn't work at any moment.

The first two (_Billy_ and _Dead Dimension_), I found that because I could not write the name of the savegames, but everything else worked (but not, because I could not use Escape key or any other key, but you don't need to use it in the game at all as everything is mouse driven). I played them with _ScummVM_ (2.8.1) and with it I could save normally, and I thought this issue was with the games themselves as their are made by the same people.

The third one (_IOWNFT_), you NEED to use the keyboard in some parts of the game, and also you need to push ESC to bring the menu to save the game. Once I realised that, I began to test things.

**Specs**

Windows 10 Pro 22H2

**The Issue** and **The "fix"**

I took the games to other computers to find the issue was not there in any of those games.

Then I began to test things in my own computer and I found two things:

1. If I execute the games as admin, the keyboard works in all of them.

2. If i close an app called _RemoTerm.exe_ that was running in my computer, the keyboard works in all of them.

So, the issue was that _RemoTerm.exe_, what was an app for controlling a Remote Control what is part of [TV Center Suite by PCTV Systems](http://www.pctvsystems.com/Support/Downloads/Application/tabid/124/language/es-ES/Default.aspx), was interfering with the version of AGS of these games. I had it installed because is the easy way to install the drivers for the DVB-T Stick (the suite itself is useless and outdated, and I not even using the remote control), so I found only the drivers (the ones in their site are blocked so I need to found them anywhere else), uninstalled the suite and the issue is gone.

But also, I thing the issue is fixed in AGS too, because with _Urban Witch Story_ ([Itch](https://postmodernadventures.itch.io/urban-witch-story)) [_AGS 3.6.0.37_], which is made by the same people of the first two I mention above (_Billy_ and _Dead..._) never had the issue with _RemoTerm.exe_ loaded, the keyboard works fine there always. So maybe between AGS _3.5.0.31_ and _3.6.0.37_ It was fixed.

By the way, with the remote control you can't do anything within the games. I tried to test it but It does nothing. And also I didn't detect any similar issue with any other game or app, only with AGS.

**To summarize**: An app _RemoTerm.exe_ from [TV Center Suite by PCTV Systems](http://www.pctvsystems.com/Support/Downloads/Application/tabid/124/language/es-ES/Default.aspx) interfere with AGS making the keyboard not work at all, until a version between _3.5.0.31_ and _3.6.0.37_ that seems to fix the issue.

Please, this report can be closed as It seems fixed already, but keeping it here could be useful for others and for future reports.

Thanks!

**Comment by ericoporto** [comment]
Between 3.5 and 3.6 the entire engine backend has been rewritten, as the internals that interface with the system went from being Allegro4 to SDL2, so all device handling was changed. The mentioned [3.6.0.37](https://github.com/adventuregamestudio/ags/releases/tag/v.3.6.0.37) was still a release candidate, that then got released at the time as [3.6.0.47](https://github.com/adventuregamestudio/ags/releases/tag/v.3.6.0.47), which was the first stable release using SDL.

I searched and the mentioned RemoTerm is not on SDL issue tracker so it probably got fixed by how SDL handles it and nothing specifically was done.

**Comment by LukasThyWalls** [comment]
Thanks @ericoporto and @ivan-mogilko , I didn't know about maybe I could use newer releases with those games, I thought there could be plenty of incompatibilities. In fact, most of these games have an EXE with all the game content (being 500 MB executable one of them), so I don't know, but If I see an issue in the future, is worth to give it a try.

The reason only to report it was to write it down somewhere, and If someone else has this issue or something similar could be useful.

Thanks!

**Comment by ericoporto** [comment]
AGS runtime can find the relevant parts on the exe and run it too. :)

**Comment by ivan-mogilko** [comment]
>  I thought there could be plenty of incompatibilities. In fact, most of these games have an EXE with all the game content (being 500 MB executable one of them), so I don't know, but If I see an issue in the future, is worth to give it a try.

AGS v3.* is almost fully backwards compatible, and can run games made with earlier versions down to v2.5. Backwards compatibility is an official engine feature of v3.* branch. This branch also shares code with the ScummVM AGS port, so we have compatibility related fixes applied as well.

It does not matter whether the game data is appended to exe or is a standalone file, the engine is capable of scanning a directory for it and reading it regardless.
